### PR TITLE
[Merged by Bors] - Resolve network interface in agent.

### LINKED ISSUE #670: "unexpected response - expected env vars response None"
### Bug Description

I tried to use a binary executable file for testing.

And the pod crashed:

```
mirrord exec --target pod/busybox-58d584f97b-pbtv8 -- ./main
✓ layer initialized
✓ agent running
  ✓ agent pod created
  ✓ pod is ready
unexpected response - expected env vars response None
Terminated
```

### Steps to Reproduce

1. Deploy a busybox image;
2. Use mirrord `mirrord exec --target pod/busybox-58d584f97b-pbtv8 -- ./main`;

### Backtrace

_No response_

### Relevant Logs

_No response_

### Your operating system and version

Centos 7.9

### Local process

None

### Local process version

_No response_

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by meowjesty**
- Network interface in agent no longer resolves to `eth0` by default (`eth0` is still used if we can't find a suitable interface);
- Fixes #670 with the new `resolve_interface` approach;

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how
 
 ## [Unreleased]
 
+### Fixed
+- mirrord-agent: No longer resolves to `eth0` by default, now we first try to resolve the appropriate network interface, if this fails then we use `eth0` as a last resource. Fixes [#670](https://github.com/metalbear-co/mirrord/issues/670).
```
Comment (aviramha) [inline CHANGELOG.md:11]: Resort*

**Review by aviramha — COMMENTED** [review]
Nitpick - can we do maybe udp instead of tcp for less possible issues?

File: CHANGELOG.md (hunk 1 above)
Comment (meowjesty) [reply CHANGELOG.md:11]: Thank you! I was going crazy trying to remember this word!

**Review by aviramha — APPROVED** [review]
Awesome work!
bors r+

**Pull request closed without merging**